Document (and check) minimum required LLVM version

We should document and possibly automatically check the required LLVM version.

```
$ pub run ffigen:setup
Building Dynamic Library for libclang wrapper...

/Users/dacoharkes/ffigen/lib/src/clang_library/wrapper.c:358:12: warning: implicit declaration of function 'clang_Cursor_isAnonymousRecordDecl' is invalid in C99 [-Wimplicit-function-declaration]
    return clang_Cursor_isAnonymousRecordDecl(*cursor);
           ^
1 warning generated.
Undefined symbols for architecture x86_64:
  "_clang_Cursor_isAnonymousRecordDecl", referenced from:
      _clang_Cursor_isAnonymousRecordDecl_wrap in wrapper-09a2d4.o
     (maybe you meant: _clang_Cursor_isAnonymousRecordDecl_wrap)
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
```

command with -v

```
$ clang -I/usr/local/opt/llvm/include/ -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/ -L/usr/local/opt/llvm/lib/ -lclang -shared -fpic /Users/dacoharkes/ffigen/lib/src/clang_library/wrapper.c -o /Users/dacoharkes/ffigen/.dart_tool/ffigen/_v_0_2_4_libwrapped_clang.dylib  -Wno-nullability-completeness -v
clang version 7.0.0 (tags/RELEASE_700/final)
Target: x86_64-apple-darwin19.5.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm/bin
 "/usr/local/Cellar/llvm/7.0.0/bin/clang-7" -cc1 -triple x86_64-apple-macosx10.15.0 -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -emit-obj -mrelax-all -disable-free -disable-llvm-verifier -discard-value-names -main-file-name wrapper.c -mrelocation-model pic -pic-level 2 -mthread-model posix -mdisable-fp-elim -masm-verbose -munwind-tables -target-cpu penryn -dwarf-column-info -debugger-tuning=lldb -target-linker-version 351.8 -v -resource-dir /usr/local/Cellar/llvm/7.0.0/lib/clang/7.0.0 -I /usr/local/opt/llvm/include/ -I /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/ -Wno-nullability-completeness -fdebug-compilation-dir /Users/dacoharkes/ffigen -ferror-limit 19 -fmessage-length 179 -stack-protector 1 -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fobjc-runtime=macosx-10.15.0 -fmax-type-align=16 -fdiagnostics-show-option -fcolor-diagnostics -o /var/folders/6l/j64crzh16j7djnq676q0994h00klzg/T/wrapper-5a8dac.o -x c /Users/dacoharkes/ffigen/lib/src/clang_library/wrapper.c
clang -cc1 version 7.0.0 based upon LLVM 7.0.0 default target x86_64-apple-darwin19.5.0
ignoring nonexistent directory "/usr/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/opt/llvm/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
 /usr/local/include
 /usr/local/Cellar/llvm/7.0.0/lib/clang/7.0.0/include
 /System/Library/Frameworks (framework directory)
 /Library/Frameworks (framework directory)
End of search list.
/Users/dacoharkes/ffigen/lib/src/clang_library/wrapper.c:358:12: warning: implicit declaration of function 'clang_Cursor_isAnonymousRecordDecl' is invalid in C99
      [-Wimplicit-function-declaration]
    return clang_Cursor_isAnonymousRecordDecl(*cursor);
           ^
1 warning generated.
 "/usr/bin/ld" -demangle -lto_library /usr/local/Cellar/llvm/7.0.0/lib/libLTO.dylib -no_deduplicate -dynamic -dylib -arch x86_64 -macosx_version_min 10.15.0 -o /Users/dacoharkes/ffigen/.dart_tool/ffigen/_v_0_2_4_libwrapped_clang.dylib -L/usr/local/opt/llvm/lib/ -lclang /var/folders/6l/j64crzh16j7djnq676q0994h00klzg/T/wrapper-5a8dac.o -lSystem /usr/local/Cellar/llvm/7.0.0/lib/clang/7.0.0/lib/darwin/libclang_rt.osx.a
Undefined symbols for architecture x86_64:
  "_clang_Cursor_isAnonymousRecordDecl", referenced from:
      _clang_Cursor_isAnonymousRecordDecl_wrap in wrapper-5a8dac.o
     (maybe you meant: _clang_Cursor_isAnonymousRecordDecl_wrap)
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
```

(llvm 7.0.0 on MacOS)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,6 @@
+# 0.2.4+1
+- Minor changes to dylib creation error log.
+
 # 0.2.4
 - Added support for C booleans as Uint8.
 - Added config `dart-bool` (default: true) to use dart bool instead of int in function parameters and return type.

--- a/bin/ffigen.dart
+++ b/bin/ffigen.dart
@@ -35,7 +35,7 @@ void main(List<String> args) {
   /// `pub run ffigen:setup -Ipath/to/llvm/include -Lpath/to/llvm/lib`.
   if (!checkDylibExist() && !autoCreateDylib()) {
     _logger.severe('Unable to create dynamic library automatically.');
-    _logger.severe('If LLVM is installed, try running:');
+    _logger.severe('If LLVM (9+) is installed, try running:');
     _logger.severe(
         '  pub run ffigen:setup -Ipath/to/llvm/include -Lpath/to/llvm/lib');
     exit(1);

--- a/bin/setup.dart
+++ b/bin/setup.dart
@@ -100,7 +100,7 @@ bool autoCreateDylib() {
   final options = _getPlatformOptions();
   final processResult = _runClangProcess(options);
   if ((processResult.stderr as String).isNotEmpty) {
-    print(stderr);
+    print(processResult.stderr);
   }
   return checkDylibExist();
 }

--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -3,7 +3,7 @@
 # BSD-style license that can be found in the LICENSE file.
 
 name: ffigen
-version: 0.2.4
+version: 0.2.4+1
 homepage: https://github.com/dart-lang/ffigen
 description: Experimental generator for FFI bindings, using LibClang to parse C header files.
 

